packages/core/test/env.test.ts

```diff
@@ -144,14 +144,31 @@ describe("createActionURL", () => {
       args: {
         action: "signout",
         protocol: undefined,
-        headers: new Headers({
-          "x-forwarded-host": "example.com",
-          "x-forwarded-proto": "https",
-        }),
-        env: { AUTH_URL: "https://env.com/api/auth/" },
-        basePath: "/auth",
+        headers: new Headers({}),
+        env: { AUTH_URL: "http://localhost:3000" },
+        basePath: "/api/auth",
+      },
+      expected: "http://localhost:3000/api/auth/signout",
+    },
+    {
+      args: {
+        action: "signout",
+        protocol: undefined,
+        headers: new Headers({}),
+        env: { AUTH_URL: "https://sub.domain.env.com" },
+        basePath: "/api/auth",
+      },
+      expected: "https://sub.domain.env.com/api/auth/signout",
+    },
+    {
+      args: {
+        action: "signout",
+        protocol: undefined,
+        headers: new Headers({}),
+        env: { AUTH_URL: "https://sub.domain.env.com/my-app/" },
+        basePath: "/api/auth",
       },
-      expected: "https://env.com/api/auth/signout",
+      expected: "https://sub.domain.env.com/my-app/api/auth/signout",
```

Comment:
This test looks weird to me. It should be a configuration error if `envUrl.pathname !== basePath`. Although we are not explicitly checking for it in the config assertion, I think we can pick either one of them and move on. I am thinking that we can prioritize the `basePath` since that's a new option. How do you think?

Reply:
This was discussed earlier with @ndom91 here https://github.com/nextauthjs/next-auth/pull/10094#issuecomment-1977096182

Arguably they env variable defines the app url+path and basepath defines the auth path. Based on earlier discussions anyway